src/pagination/index.tsx

```diff
@@ -84,105 +109,167 @@ export default factory(function Pagination({ middleware: { theme, icache, i18n }
 
 	if (pageSizes !== undefined && pageSizes !== icache.get('pageSizes')) {
 		icache.set('pageSizes', pageSizes);
-		icache.set('pageSizesData', pageSizes.map((ps) => ({ value: ps.toString() })));
+		icache.set(
+			'pageSizesResource',
+			createMemoryResourceWithData(pageSizes.map((ps) => ({ value: ps.toString() })))
+		);
 	}
 
 	const page = icache.getOrSet('currentPage', 1);
 	const pageSize = icache.getOrSet('pageSize', 10);
-	const pageSizesData = icache.get('pageSizesData');
+	const pageSizesResource = icache.get('pageSizesResource');
+
+	const getRenderedWidth = (dnode: RenderResult) => {
```

Comment:
this should be outside of the factory so it's only created once

Reply:
I had it outside, but it needs `theme.variant()` to render properly. I could just pass that in for each case.